### ISSUE TITLE
'Unexpected variable NaN' during compilation

### ISSUE THREAD

**Issue by abowsher** [bertramdev/grails-asset-pipeline#315]
We had a javascript file with suspicious formatting.  It worked fine in the browser, but gave the Asset Pipeline compiler trouble (more in a moment).  The bigger issue is the error generated by the Closure Compiler.  Since it's buried under Asset Pipeline, I'm not sure how best to fix it (I was able to fix the javascript to get it to pass, but I mean how to fix the Closure Compiler).

Basically, it gets the following error (see the bottom for the full stacktrace):

ERROR pipeline.AssetCompiler  - Uglify JS Exception
Message: INTERNAL COMPILER ERROR.
Please report this problem.

```
Unexpected variable NaN
  Node(NAME NaN): application.unminified.js:15200:13
            +'</div></div></li>';
  Parent(COMMA): application.unminified.js:15200:12
            +'</div></div></li>';
... (truncated) ...
Caused by IllegalStateException: Unexpected variable NaN
->>  198 | visit            in com.google.javascript.jscomp.VarCheck
```

It sounds similar to these two issues:
https://code.google.com/p/closure-compiler/issues/detail?id=776
https://code.google.com/p/closure-compiler/issues/detail?id=1031

The original JavaScript that triggers the error looks like this (yes I know it's ugly but wanted to help you reproduce it):
```
var bar;
var length = (value / max) * 100;
bar = '<li id="' + key.toLowerCase().replace(' ', '-') + '"><div class="key">' + key.toLowerCase() + '</div><div class="bar-holder"><div class="bar" style="color: rgb(101, 101, 101);">' + Number(value).toLocaleString('en');
    +'</div></div></li>';
```

Here's the full stacktrace:
```
|Processing File 95 of 1031 - application.js
|Uglifying File 95 of 1031 - application
Error |
2015-12-10 09:13:00,776 [main] ERROR pipeline.AssetCompiler  - Uglify JS Exception
Message: INTERNAL COMPILER ERROR.
Please report this problem.

Unexpected variable NaN
  Node(NAME NaN): application.unminified.js:15200:13
            +'</div></div></li>';
  Parent(COMMA): application.unminified.js:15200:12
            +'</div></div></li>';

    Line | Method
->>  198 | visit            in com.google.javascript.jscomp.VarCheck
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|    587 | traverseBranch   in com.google.javascript.jscomp.NodeTraversal
|    581 | traverseBranch . in     ''
|    621 | traverseFunction in     ''
|    572 | traverseBranch . in     ''
|    581 | traverseBranch   in     ''
|    621 | traverseFunction in     ''
|    572 | traverseBranch   in     ''
|    581 | traverseBranch . in     ''
|    316 | traverseRoots    in     ''
|    140 | process . . . .  in com.google.javascript.jscomp.VarCheck
|    285 | process          in com.google.javascript.jscomp.PhaseOptimizer$NamedPass
|    217 | process . . . .  in com.google.javascript.jscomp.PhaseOptimizer
|   1987 | optimize         in com.google.javascript.jscomp.Compiler
|    769 | compileInternal  in     ''
|     93 | access$000       in     ''
|    655 | call . . . . . . in com.google.javascript.jscomp.Compiler$3
|    652 | call             in     ''
|    699 | call . . . . . . in com.google.javascript.jscomp.Compiler$4
|    334 | innerRun         in java.util.concurrent.FutureTask$Sync
|    166 | run . . . . . .  in java.util.concurrent.FutureTask
|   1110 | runWorker        in java.util.concurrent.ThreadPoolExecutor
|    603 | run . . . . . .  in java.util.concurrent.ThreadPoolExecutor$Worker
^    722 | run              in java.lang.Thread

Caused by IllegalStateException: Unexpected variable NaN
->>  198 | visit            in com.google.javascript.jscomp.VarCheck
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|    587 | traverseBranch   in com.google.javascript.jscomp.NodeTraversal
|    581 | traverseBranch . in     ''
|    621 | traverseFunction in     ''
|    572 | traverseBranch . in     ''
|    581 | traverseBranch   in     ''
|    621 | traverseFunction in     ''
|    572 | traverseBranch   in     ''
|    581 | traverseBranch . in     ''
|    316 | traverseRoots    in     ''
|    140 | process . . . .  in com.google.javascript.jscomp.VarCheck
|    285 | process          in com.google.javascript.jscomp.PhaseOptimizer$NamedPass
|    217 | process . . . .  in com.google.javascript.jscomp.PhaseOptimizer
|   1987 | optimize         in com.google.javascript.jscomp.Compiler
|    769 | compileInternal  in     ''
|     93 | access$000       in     ''
|    655 | call . . . . . . in com.google.javascript.jscomp.Compiler$3
|    652 | call             in     ''
|    699 | call . . . . . . in com.google.javascript.jscomp.Compiler$4
|    334 | innerRun         in java.util.concurrent.FutureTask$Sync
|    166 | run . . . . . .  in java.util.concurrent.FutureTask
|   1110 | runWorker        in java.util.concurrent.ThreadPoolExecutor
|    603 | run . . . . . .  in java.util.concurrent.ThreadPoolExecutor$Worker
^    722 | run              in java.lang.Thread
```

**Comment by davydotcom** [comment]
If it's a closure but not much I can do, but I do have a build staged with an updated version of closure

**Comment by davydotcom** [comment]
new closure compiler build released in 2.7.0
